library/azure_rm_securitygroup.py

```diff
@@ -425,6 +445,10 @@ def compare_rules(old_rule, rule):
         changed = True
     if set(rule.get('destination_port_ranges') or []) != set(old_rule.get('destination_port_ranges') or []):
         changed = True
+    if set(rule.get('source_application_security_groups') or []) != set(old_rule.get('source_application_security_groups') or []):
```

Comment:
you can just do "rule.get('source_application_security_groups', [])"
I think it' simpler

Reply:
copy from above, will keep it